UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterTile.cs

```diff
@@ -75,51 +75,57 @@ public NetworkInstanceId ParentNetId
 	/// </summary>
 	public Vector3Int WorldPosition => isServer ? WorldPositionServer : WorldPositionClient;
 
-	public Vector3Int WorldPositionServer => MatrixManager.Instance.LocalToWorldInt(serverPosition, Matrix);
-	public Vector3Int WorldPositionClient => MatrixManager.Instance.LocalToWorldInt(clientPosition, Matrix);
+	public Vector3Int WorldPositionServer => MatrixManager.Instance.LocalToWorldInt(serverLocalPosition, Matrix);
+	public Vector3Int WorldPositionClient => MatrixManager.Instance.LocalToWorldInt(clientLocalPosition, Matrix);
+
+	/// <summary>
+	/// Registered local position of this object. Returns correct value depending on if this is on the
+	/// server or client.
+	/// </summary>
+	public Vector3Int LocalPosition => isServer ? LocalPositionServer : LocalPositionClient;
```

Comment:
I'm not sure if this one is going to be reliable, it's proven to be the best practice to request either client or server value explicitly, otherwise debugging becomes a pain in the ass